### DISCUSSION TITLE
Getting a Async Promise Element array object from selector?

### DISCUSSION THREAD

**Discussion by TheRealCleany** [webdriverio/webdriverio#7933] [Q&A]
Hello!

I started to transpose my sync code to async code and I'm stuck at arrays...I'm using a selector to get multiple elements, but it doesn't work anymore in async mode...anyone have an idea to fix that? Thusfar I've got this:

get selectedQuestions(): ChainablePromiseArray<Promise<WebdriverIO.Element>> { return $$('[name="SelectedQuestions"]:checked')}

Old code:
get selectedQuestions(): WebdriverIO.Element[] {return $$([name="SelectedQuestions"]:checked)}

I also tried:
get selectedQuestions(): ChainablePromiseElement<Promise<WebdriverIO.Element>>[] { return $$('[name="SelectedQuestions"]:checked')

Error message on return element:  "Type 'ChainablePromiseArray<ElementArray>' is not assignable to type 'ChainablePromiseArray<Promise<Element>>'.
Type 'ElementArray' is missing the following properties from type 'Promise<Element>': then, catch, finally,"

**Comment by christian-bromann** [comment]
Just type it as TypeScript suggests it:

```ts
get selectedQuestions(): ChainablePromiseArray<Promise<WebdriverIO.Element>> { return $$('[name="SelectedQuestions"]:checked')
```

**Reply by TheRealCleany** [reply]
Thanks
